eureka/S5_lightcurve_fitting/fitters.py

```diff
@@ -125,7 +170,428 @@ def demcfitter(time, data, model, uncertainty=None, verbose=True, **kwargs):
+
+    if meta.isplots_S5 > 3:
+        rmsplot(lc, model_lc, meta, figname='allanplot_emcee.png')
+
+    return best_model
+
+
+def dynestyfitter(lc, model, meta, **kwargs):
+    """Perform sampling using emcee
+
+    Parameters
+    ----------
+    data: sequence
+        The observational data
+    model: ExoCTK.lightcurve_fitting.models.Model
+        The model to fit
+    uncertainty: np.ndarray (optional)
+        The uncertainty on the (same shape) data
+    method: str
+        The name of the method to use
+    name: str
+        A name for the best fit model
+    verbose: bool
+        Print some stuff
+
+    Returns
+    -------
+    lsq.Model.fit.fit_report
+        The results of the fit
+    """
+    # Concatenate the lists of parameters
+    #all_keys   = [i for j in [model.components[n].parameters.dict.keys()
+    #              for n in range(len(model.components))] for i in j]
+
+    # RUN LEAST SQUARES
+    lsq_sol = lsqfitter(lc, model, meta, **kwargs)
+    print(lsq_sol)
+    # SCALE UNCERTAINTIES WITH REDUCED CHI2 TODO: put a flag for that into config
+    lc.unc *= np.sqrt(lsq_sol.chi2red)
+
+    all_params = [i for j in [model.components[n].parameters.dict.items()
+                  for n in range(len(model.components))] for i in j]
+
+    #print(all_params)
+    # Group the different variable types
+    freenames = []
+    freepars = []
+    pmin = []
+    pmax = []
+    indep_vars = {}
+    for ii, item in enumerate(all_params):
+        name, param = item
+        #param = list(param)
+        if param[1] == 'free':
+            freenames.append(name)
+            freepars.append(param[0])
+            if len(param) > 3:
+                pmin.append(param[2])
+                pmax.append(param[3])
+            else:
+                pmin.append(-np.inf)
+                pmax.append(np.inf)
+        # elif param[1] == 'fixed':
+        #     pinitial.append(param[0])
+        #     pmin.append(param[0])
+        #     pmax.append(param[0])
+        elif param[1] == 'independent':
+            indep_vars[name] = param[0]
+    freepars = np.array(freepars)
+    pmin = np.array(pmin)
+    pmax = np.array(pmax)
+    print('before update: ', freepars)
+
+    # UPDATE MODEL PARAMETERS WITH LSQ BEST FIT
+    model.update(lsq_sol.fit_params, freenames)
+    print('after update: ', freepars)
+
+    # Set the uncertainty
+    if lc.unc is None:
+        lc.unc = np.sqrt(lc.flux)
+
+    #lc.etc = {}
+    #lc.etc['time'] = lc.time
+    import time
+
+
+    # DYNESTY
+
+    #PRIOR TRANSFORMATION TODO: ADD GAUSSIAN PRIORS
+    def transform_uniform(x, a, b):
+        return a + (b - a) * x
+
+    def transform_normal(x, mu, sigma):
+        return norm.ppf(x, loc=mu, scale=sigma)
+
+    def ptform(theta):
+        p = np.zeros_like(theta)
+        n = len(theta)
+        for i in range(n):
+            p[i] = transform_uniform(theta[i], pmin[i], pmax[i])
+        return p
+
+
+    def ln_like(theta, lc, model, pmin, pmax):
```

Comment:
Let's move all of the log likelihood functions out of `fitters.py` and into a separate file

Reply:
Now addressed in s5_tjbEdits (moved to a new likelihood.py)